src/main/java/com/googleintern/wfm/ruleengine/action/RuleValidation.java

```diff
@@ -0,0 +1,286 @@
+package src.main.java.com.googleintern.wfm.ruleengine.action;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableSetMultimap;
+import com.google.common.collect.Sets;
+import com.opencsv.CSVWriter;
+import src.main.java.com.googleintern.wfm.ruleengine.model.FilterModel;
+import src.main.java.com.googleintern.wfm.ruleengine.model.PoolAssignmentModel;
+import src.main.java.com.googleintern.wfm.ruleengine.model.RuleModel;
+import src.main.java.com.googleintern.wfm.ruleengine.model.UserPoolAssignmentModel;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * RuleValidation class is used to test the performance of generated rules and write validation
+ * results into a csv file.
+ */
+public class RuleValidation {
+  enum Separator {
+    COMMA(","),
+    SQUARE_BRACKET_LEFT("["),
+    SQUARE_BRACKET_RIGHT("]"),
+    CURLY_BRACKET_RIGHT("}"),
```

Comment:
Just curious that we only need a CURLY_BRACKET_RIGHT and we never need a CURLY_BRACKET_LEFT?

Reply:
I included "{" in a constant string variable `CASE_POOL_ID_PREFIX` before. I will add "{" to the enum and replace it the constant string.